src/blocks/frontend/live-search/index.ts

```diff
@@ -246,4 +275,19 @@ domReady( () => {
 			container.removeChild( loading as Node );
 		}
 	};
+
+	const highlight = ( element: HTMLElement, input: Element ) => {
+		element.classList.add( 'highlight' );
+
+		const inputBackground = getComputedStyle( input ).backgroundColor;
+		const hsl = rgb2hsl( inputBackground );
+		const dark = 50 >= hsl[2];
+
+		element.style.backgroundColor = dark ? 'rgba(255, 255, 255, 0.12)' : 'rgba(49, 50, 51, 0.12)';
```

Comment:
Is this kind of a dark mode detector?

Reply:
yes, it is 🙈

Reply:
Let's make it more specific. I wondered a bit why did you make a transformation to HSL.

```javascript
// Determine the background color for a light/dark theme
const inputBackground = getComputedStyle( input ).backgroundColor;
const [,,lightness] = rgb2hsl( inputBackground );
const isDarkBackground = 50 >= lightness;
```